account-kit/react/src/components/auth/types.ts

```diff
@@ -9,4 +9,5 @@ export type AuthType =
       placeholder?: string;
     }
   | { type: "passkey" }
-  | { type: "external_wallets"; walletConnect?: WalletConnectParameters };
+  | { type: "external_wallets"; walletConnect?: WalletConnectParameters }
+  | { type: "social"; googleAuth: boolean };
```

Comment:
nit: I think this type should look more like

```ts
{ type: "social" } & (
  | {
      // this is auth0 for things we don't provide out of the box
      authProviderId: "auth0";
      isCustomProvider?: false;
      auth0Connection?: string;
      logoUrl: string;
    }
  | {
      // this one handles things like google
      authProviderId: KnownAuthProvider;
      isCustomProvider?: false;
      auth0Connection?: never;
    }
  | {
      // this is truly custom auth, we can probably omit this one all together
      authProviderId: string;
      isCustomProvider: true;
      auth0Connection?: never;
      logoUrl: string;
    }
) & OauthProviderConfig &
      OauthRedirectConfig;
```

Reply:
the type above is basically the same as `oauth` AuthParams type, we may want to consolidate them

We will also need to handle the case where `type.mode === "popup"` in the `createConfig` function to make sure that if this is set, we always set the `enablePopupOauth` param to true regardless of what the dev passes in here since it's required